stdlib/seq.mli

```diff
@@ -13,97 +13,760 @@
+
+    @raise Forced_twice if [once xs], or a suffix of it,
+           is queried more than once.
+
+    @since 4.14 *)
+
+val transpose : 'a t t -> 'a t t
+(** If [xss] is a matrix (a sequence of rows), then [transpose xss] is
+    the sequence of the columns of the matrix [xss].
+
+    The rows of the matrix [xss] are not required to have the same length.
+
+    The matrix [xss] is not required to be finite (in either direction).
+
+    The matrix [xss] must be persistent.
+
+    @since 4.14 *)
+
+(** {1 Combining sequences} *)
+
+val append : 'a t -> 'a t -> 'a t
+(** [append xs ys] is the concatenation of the sequences [xs] and [ys].
+
+    Its elements are the elements of [xs], followed with the elements of [ys].
+
+    @since 4.11 *)
 
 val concat : 'a t t -> 'a t
-(** concatenate a sequence of sequences.
+(** If [xss] is a sequence of sequences,
+    then [concat xss] is its concatenation.
 
-    @since 4.13
- *)
+    If [xss] is the sequence [xs0; xs1; ...] then
+    [concat xss] is the sequence [xs0 @ xs1 @ ...].
+
+    @since 4.13 *)
 
 val flat_map : ('a -> 'b t) -> 'a t -> 'b t
-(** Map each element to a subsequence, then return each element of this
-    sub-sequence in turn.
-    This transformation is lazy, it only applies when the result is
-    traversed. *)
+(** [flat_map f xs] is equivalent to [concat (map f xs)]. *)
 
 val concat_map : ('a -> 'b t) -> 'a t -> 'b t
-(** Alias for {!flat_map}.
+(** [concat_map f xs] is equivalent to [concat (map f xs)].
 
-    @since 4.13
-*)
+    [concat_map] is an alias for [flat_map].
 
-val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b t -> 'a
-(** Traverse the sequence from left to right, combining each element with the
-    accumulator using the given function.
-    The traversal happens immediately and will not terminate on infinite
-    sequences.
+    @since 4.13 *)
 
-    Also see {!List.fold_left} *)
+val zip : 'a t -> 'b t -> ('a * 'b) t
+(** [zip xs ys] is the sequence of pairs [(x, y)]
+    drawn synchronously from the sequences [xs] and [ys].
 
-val iter : ('a -> unit) -> 'a t -> unit
-(** Iterate on the sequence, calling the (imperative) function on every element.
-    The traversal happens immediately and will not terminate on infinite
-    sequences. *)
+    If the sequences [xs] and [ys] have different lengths, then
+    the sequence ends as soon as one sequence is exhausted;
+    the excess elements in the other sequence are ignored.
 
-val unfold : ('b -> ('a * 'b) option) -> 'b -> 'a t
-(** Build a sequence from a step function and an initial value.
-    [unfold f u] returns [empty] if [f u] returns [None],
-    or [fun () -> Cons (x, unfold f y)] if [f u] returns [Some (x, y)].
+    [zip xs ys] is equivalent to [map2 (fun a b -> (a, b)) xs ys].
 
-    For example, [unfold (function [] -> None | h::t -> Some (h,t)) l]
-    is equivalent to [List.to_seq l].
-    @since 4.11 *)
+    @since 4.14 *)
+
+val combine : 'a t -> 'b t -> ('a * 'b) t
+(** [combine] is an alias for [zip].
```

Comment:
Regarding the semantics of `combine`, I agree that it could be confusing if it does not have the same semantics as `List.combine`. My suggestion in this case is to just remove `combine` for now, keep `zip`, and decide later whether we want a function named `combine` and with what semantics.

Reply:
So, I am removing `combine`.